

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>The biom file format &mdash; biom-format.org</title>
    <link rel="stylesheet" href="../_static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../_static/print.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.9.3',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <script type="text/javascript" src="../_static/theme_extras.js"></script>
    <link rel="top" title="biom-format.org" href="../index.html" />
    <link rel="up" title="BIOM Documentation" href="index.html" />
    <link rel="next" title="biom-format Table objects" href="table_objects.html" />
    <link rel="prev" title="BIOM Documentation" href="index.html" /> 
  </head>
  <body>
      <div class="header"><h1 class="heading"><a href="../index.html">
          <span>biom-format.org</span></a></h1>
        <h2 class="heading"><span>The biom file format</span></h2>
      </div>
      <div class="topnav">
      
        <p>
        «&#160;&#160;<a href="index.html">BIOM Documentation</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="table_objects.html">biom-format <tt class="docutils literal docutils literal"><span class="pre">Table</span></tt> objects</a>&#160;&#160;»
        </p>

      </div>
      <div class="content">
        
        
  <div class="section" id="the-biom-file-format">
<span id="biom-format"></span><h1>The biom file format<a class="headerlink" href="#the-biom-file-format" title="Permalink to this headline">¶</a></h1>
<p>The <tt class="docutils literal"><span class="pre">biom</span></tt> format is based on <a class="reference external" href="http://www.json.org">JSON</a> to provide the overall structure for the format. JSON is a widely supported format with native parsers available within many programming languages.</p>
<p>Required top-level fields:</p>
<div class="highlight-python"><pre>id                  : &lt;string or null&gt; a field that can be used to id a table (or null)
format              : &lt;string&gt; The name and version of the current biom format
format_url          : &lt;url&gt; A string with a static URL providing format details
type                : &lt;string&gt; Table type (a controlled vocabulary)
                      Acceptable values:
                       "OTU table"
                       "Pathway table"
                       "Function table"
                       "Ortholog table"
                       "Gene table"
                       "Metabolite table"
                       "Taxon table"
generated_by        : &lt;string&gt; Package and revision that built the table
date                : &lt;datetime&gt; Date the table was built (ISO 8601 format)
rows                : &lt;list of objects&gt; An ORDERED list of obj describing the rows
                      (explained in detail below)
columns             : &lt;list of objects&gt; An ORDERED list of obj  describing the columns
                      (explained in detail below)
matrix_type         : &lt;string&gt; Type of matrix data representation (a controlled vocabulary)
                      Acceptable values:
                       "sparse" : only non-zero values are specified
                       "dense" : every element must be specified
matrix_element_type : Value type in matrix (a controlled vocabulary)
                      Acceptable values:
                       "int" : integer
                       "float" : floating point
                       "unicode" : unicode string
shape               : &lt;list of ints&gt;, the number of rows and number of columns in data
data                : &lt;list of lists&gt;, counts of observations by sample
                       if matrix_type is "sparse", [[row, column, value],
                                                    [row, column, value],
                                                    ...]
                       if matrix_type is "dense",  [[value, value, value, ...],
                                                    [value, value, value, ...],
                                                    ...]</pre>
</div>
<p>Optional top-level fields:</p>
<div class="highlight-python"><pre>comment             : &lt;string&gt; A free text field containing any information that you
                       feel is relevant (or just feel like sharing)</pre>
</div>
<p>The rows value is an ORDERED list of objects where each object corresponds to a single
row in the matrix. Each object can currently store arbitrary keys, although
this might become restricted based on table type. Each object must provide,
at the minimum:</p>
<div class="highlight-python"><pre>id                  : &lt;string&gt; an arbitrary UNIQUE identifier
metadata            : &lt;an object or null&gt; A object containing key, value metadata pairs</pre>
</div>
<p>The columns value is an ORDERED list of objects where each object corresponds to a single
column in the matrix. Each object can currently store arbitrary keys, although
this might become restricted based on table type. Each object must provide,
at the minimum:</p>
<div class="highlight-python"><pre>id                  : &lt;string&gt; an arbitrary UNIQUE identifier
metadata            : &lt;an object or null&gt; A object containing key, value metadata pairs</pre>
</div>
<div class="section" id="example-biom-files">
<h2>Example biom files<a class="headerlink" href="#example-biom-files" title="Permalink to this headline">¶</a></h2>
<p>Below are examples of minimal and rich biom files in both sparse and dense formats. To decide which of these you should generate for new data types, see the section on <a class="reference internal" href="#sparse-or-dense"><em>Should I generate sparse or dense biom files?</em></a>.</p>
<div class="section" id="minimal-sparse-otu-table">
<h3>Minimal sparse OTU table<a class="headerlink" href="#minimal-sparse-otu-table" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre><span class="p">{</span>
    <span class="s">&quot;id&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">,</span>
    <span class="s">&quot;format&quot;</span><span class="p">:</span> <span class="s">&quot;Biological Observation Matrix 0.9.1-dev&quot;</span><span class="p">,</span>
    <span class="s">&quot;format_url&quot;</span><span class="p">:</span> <span class="s">&quot;http://biom-format.org&quot;</span><span class="p">,</span>
    <span class="s">&quot;type&quot;</span><span class="p">:</span> <span class="s">&quot;OTU table&quot;</span><span class="p">,</span>
    <span class="s">&quot;generated_by&quot;</span><span class="p">:</span> <span class="s">&quot;QIIME revision 1.4.0-dev&quot;</span><span class="p">,</span>
    <span class="s">&quot;date&quot;</span><span class="p">:</span> <span class="s">&quot;2011-12-19T19:00:00&quot;</span><span class="p">,</span>
    <span class="s">&quot;rows&quot;</span><span class="p">:[</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">}</span>
        <span class="p">],</span>
    <span class="s">&quot;columns&quot;</span><span class="p">:</span> <span class="p">[</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample6&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">}</span>
        <span class="p">],</span>
    <span class="s">&quot;matrix_type&quot;</span><span class="p">:</span> <span class="s">&quot;sparse&quot;</span><span class="p">,</span>
    <span class="s">&quot;matrix_element_type&quot;</span><span class="p">:</span> <span class="s">&quot;int&quot;</span><span class="p">,</span>
    <span class="s">&quot;shape&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">],</span>
    <span class="s">&quot;data&quot;</span><span class="p">:[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">4</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
            <span class="p">[</span><span class="mi">4</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
           <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="section" id="minimal-dense-otu-table">
<h3>Minimal dense OTU table<a class="headerlink" href="#minimal-dense-otu-table" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre><span class="p">{</span>
    <span class="s">&quot;id&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">,</span>
    <span class="s">&quot;format&quot;</span><span class="p">:</span> <span class="s">&quot;Biological Observation Matrix 0.9.1-dev&quot;</span><span class="p">,</span>
    <span class="s">&quot;format_url&quot;</span><span class="p">:</span> <span class="s">&quot;http://biom-format.org&quot;</span><span class="p">,</span>
    <span class="s">&quot;type&quot;</span><span class="p">:</span> <span class="s">&quot;OTU table&quot;</span><span class="p">,</span>
    <span class="s">&quot;generated_by&quot;</span><span class="p">:</span> <span class="s">&quot;QIIME revision 1.4.0-dev&quot;</span><span class="p">,</span>
    <span class="s">&quot;date&quot;</span><span class="p">:</span> <span class="s">&quot;2011-12-19T19:00:00&quot;</span><span class="p">,</span>
    <span class="s">&quot;rows&quot;</span><span class="p">:[</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">}</span>
        <span class="p">],</span>
    <span class="s">&quot;columns&quot;</span><span class="p">:</span> <span class="p">[</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">},</span>
            <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample6&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">}</span>
        <span class="p">],</span>
    <span class="s">&quot;matrix_type&quot;</span><span class="p">:</span> <span class="s">&quot;dense&quot;</span><span class="p">,</span>
    <span class="s">&quot;matrix_element_type&quot;</span><span class="p">:</span> <span class="s">&quot;int&quot;</span><span class="p">,</span>
    <span class="s">&quot;shape&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span>
    <span class="s">&quot;data&quot;</span><span class="p">:</span>  <span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">],</span>
              <span class="p">[</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
              <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">0</span><span class="p">],</span>
              <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
              <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">]]</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="section" id="rich-sparse-otu-table">
<h3>Rich sparse OTU table<a class="headerlink" href="#rich-sparse-otu-table" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre><span class="p">{</span>
 <span class="s">&quot;id&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">,</span>
 <span class="s">&quot;format&quot;</span><span class="p">:</span> <span class="s">&quot;Biological Observation Matrix 0.9.1-dev&quot;</span><span class="p">,</span>
 <span class="s">&quot;format_url&quot;</span><span class="p">:</span> <span class="s">&quot;http://biom-format.org&quot;</span><span class="p">,</span>
 <span class="s">&quot;type&quot;</span><span class="p">:</span> <span class="s">&quot;OTU table&quot;</span><span class="p">,</span>
 <span class="s">&quot;generated_by&quot;</span><span class="p">:</span> <span class="s">&quot;QIIME revision 1.4.0-dev&quot;</span><span class="p">,</span>
 <span class="s">&quot;date&quot;</span><span class="p">:</span> <span class="s">&quot;2011-12-19T19:00:00&quot;</span><span class="p">,</span>
 <span class="s">&quot;rows&quot;</span><span class="p">:[</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Proteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Gammaproteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;o__Enterobacteriales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Enterobacteriaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Escherichia&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Cyanobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Nostocophycideae&quot;</span><span class="p">,</span> <span class="s">&quot;o__Nostocales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Nostocaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Dolichospermum&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Archaea&quot;</span><span class="p">,</span> <span class="s">&quot;p__Euryarchaeota&quot;</span><span class="p">,</span> <span class="s">&quot;c__Methanomicrobia&quot;</span><span class="p">,</span> <span class="s">&quot;o__Methanosarcinales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Methanosarcinaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Methanosarcina&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Firmicutes&quot;</span><span class="p">,</span> <span class="s">&quot;c__Clostridia&quot;</span><span class="p">,</span> <span class="s">&quot;o__Halanaerobiales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Halanaerobiaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Halanaerobium&quot;</span><span class="p">,</span> <span class="s">&quot;s__Halanaerobiumsaccharolyticum&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Proteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Gammaproteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;o__Enterobacteriales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Enterobacteriaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Escherichia&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}}</span>
    <span class="p">],</span>
 <span class="s">&quot;columns&quot;</span><span class="p">:[</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CGCTTATCGAGA&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATACCAGTAGC&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCTACCTGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCGGCCTGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCTACCAAT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample6&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTAACTACCAAT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}}</span>
            <span class="p">],</span>
 <span class="s">&quot;matrix_type&quot;</span><span class="p">:</span> <span class="s">&quot;sparse&quot;</span><span class="p">,</span>
 <span class="s">&quot;matrix_element_type&quot;</span><span class="p">:</span> <span class="s">&quot;int&quot;</span><span class="p">,</span>
 <span class="s">&quot;shape&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">],</span>
 <span class="s">&quot;data&quot;</span><span class="p">:[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">5</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">3</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">1</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">4</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">3</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">4</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
         <span class="p">[</span><span class="mi">4</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">]</span>
        <span class="p">]</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
<div class="section" id="rich-dense-otu-table">
<h3>Rich dense OTU table<a class="headerlink" href="#rich-dense-otu-table" title="Permalink to this headline">¶</a></h3>
<div class="highlight-python"><div class="highlight"><pre><span class="p">{</span>
 <span class="s">&quot;id&quot;</span><span class="p">:</span><span class="n">null</span><span class="p">,</span>
 <span class="s">&quot;format&quot;</span><span class="p">:</span> <span class="s">&quot;Biological Observation Matrix 0.9.1-dev&quot;</span><span class="p">,</span>
 <span class="s">&quot;format_url&quot;</span><span class="p">:</span> <span class="s">&quot;http://biom-format.org&quot;</span><span class="p">,</span>
 <span class="s">&quot;type&quot;</span><span class="p">:</span> <span class="s">&quot;OTU table&quot;</span><span class="p">,</span>
 <span class="s">&quot;generated_by&quot;</span><span class="p">:</span> <span class="s">&quot;QIIME revision 1.4.0-dev&quot;</span><span class="p">,</span>
 <span class="s">&quot;date&quot;</span><span class="p">:</span> <span class="s">&quot;2011-12-19T19:00:00&quot;</span><span class="p">,</span>
 <span class="s">&quot;rows&quot;</span><span class="p">:[</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Proteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Gammaproteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;o__Enterobacteriales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Enterobacteriaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Escherichia&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Cyanobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Nostocophycideae&quot;</span><span class="p">,</span> <span class="s">&quot;o__Nostocales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Nostocaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Dolichospermum&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Archaea&quot;</span><span class="p">,</span> <span class="s">&quot;p__Euryarchaeota&quot;</span><span class="p">,</span> <span class="s">&quot;c__Methanomicrobia&quot;</span><span class="p">,</span> <span class="s">&quot;o__Methanosarcinales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Methanosarcinaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Methanosarcina&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Firmicutes&quot;</span><span class="p">,</span> <span class="s">&quot;c__Clostridia&quot;</span><span class="p">,</span> <span class="s">&quot;o__Halanaerobiales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Halanaerobiaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Halanaerobium&quot;</span><span class="p">,</span> <span class="s">&quot;s__Halanaerobiumsaccharolyticum&quot;</span><span class="p">]}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;GG_OTU_5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span><span class="s">&quot;taxonomy&quot;</span><span class="p">:[</span><span class="s">&quot;k__Bacteria&quot;</span><span class="p">,</span> <span class="s">&quot;p__Proteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;c__Gammaproteobacteria&quot;</span><span class="p">,</span> <span class="s">&quot;o__Enterobacteriales&quot;</span><span class="p">,</span> <span class="s">&quot;f__Enterobacteriaceae&quot;</span><span class="p">,</span> <span class="s">&quot;g__Escherichia&quot;</span><span class="p">,</span> <span class="s">&quot;s__&quot;</span><span class="p">]}}</span>
    <span class="p">],</span>
 <span class="s">&quot;columns&quot;</span><span class="p">:[</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample1&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CGCTTATCGAGA&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample2&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATACCAGTAGC&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample3&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCTACCTGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;gut&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human gut&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample4&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCGGCCTGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample5&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTCTCTACCAAT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}},</span>
    <span class="p">{</span><span class="s">&quot;id&quot;</span><span class="p">:</span><span class="s">&quot;Sample6&quot;</span><span class="p">,</span> <span class="s">&quot;metadata&quot;</span><span class="p">:{</span>
                             <span class="s">&quot;BarcodeSequence&quot;</span><span class="p">:</span><span class="s">&quot;CTAACTACCAAT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;LinkerPrimerSequence&quot;</span><span class="p">:</span><span class="s">&quot;CATGCTGCCTCCCGTAGGAGT&quot;</span><span class="p">,</span>
                             <span class="s">&quot;BODY_SITE&quot;</span><span class="p">:</span><span class="s">&quot;skin&quot;</span><span class="p">,</span>
                             <span class="s">&quot;Description&quot;</span><span class="p">:</span><span class="s">&quot;human skin&quot;</span><span class="p">}}</span>
            <span class="p">],</span>
 <span class="s">&quot;matrix_type&quot;</span><span class="p">:</span> <span class="s">&quot;dense&quot;</span><span class="p">,</span>
 <span class="s">&quot;matrix_element_type&quot;</span><span class="p">:</span> <span class="s">&quot;int&quot;</span><span class="p">,</span>
 <span class="s">&quot;shape&quot;</span><span class="p">:</span> <span class="p">[</span><span class="mi">5</span><span class="p">,</span><span class="mi">6</span><span class="p">],</span>
 <span class="s">&quot;data&quot;</span><span class="p">:</span>  <span class="p">[[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">],</span>
           <span class="p">[</span><span class="mi">5</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
           <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">4</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">0</span><span class="p">],</span>
           <span class="p">[</span><span class="mi">2</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">],</span>
           <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="mi">0</span><span class="p">]]</span>
<span class="p">}</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="should-i-generate-sparse-or-dense-biom-files">
<span id="sparse-or-dense"></span><h2>Should I generate sparse or dense biom files?<a class="headerlink" href="#should-i-generate-sparse-or-dense-biom-files" title="Permalink to this headline">¶</a></h2>
<p>In general, we recommend using the sparse format for your biom files. These will be a lot smaller than the dense format biom files when your data is sparse (i.e., more than 85% of your counts are zero). This is common for OTU tables and metagenome tables, and you&#8217;ll want to investigate whether it&#8217;s true for your data. If you currently format your data in tab-separated tables where observations are rows and samples are columns, you can format that file to be convertible to biom format with the <tt class="docutils literal"><span class="pre">convert_biom.py</span></tt>. Here you can create dense and sparse formats, and see which file size is smaller. See the section on <a class="reference internal" href="biom_conversion.html#converting"><em>Converting between file formats</em></a>.</p>
</div>
<div class="section" id="motivation-for-the-biom-format">
<h2>Motivation for the BIOM format<a class="headerlink" href="#motivation-for-the-biom-format" title="Permalink to this headline">¶</a></h2>
<p>The BIOM format was motivation by several goals. First, to facilitate efficient handling and storage of large, sparse biological contingency tables; second, to support encapsulation of core study data (contingency table data and sample/observation metadata) in a single file; and third, to facilitate the use of these tables between tools that support this format (e.g., passing of data between <a class="reference external" href="http://www.qiime.org">QIIME</a>, <a class="reference external" href="http://metagenomics.anl.gov">MG-RAST</a>, and <a class="reference external" href="http://vamps.mbl.edu/">VAMPS</a>.).</p>
<div class="section" id="efficient-handling-and-storage-of-very-large-tables">
<h3>Efficient handling and storage of very large tables<a class="headerlink" href="#efficient-handling-and-storage-of-very-large-tables" title="Permalink to this headline">¶</a></h3>
<p>In QIIME, we began hitting limitations with OTU table objects when working with thousands of samples and hundreds of thousands of OTUs. In the near future we expect that we&#8217;ll be dealing with hundreds of thousands of samples in single analyses.</p>
<p>The OTU table format up to QIIME 1.4.0 involved a dense matrix: if an OTU was not observed in a given sample, that would be indicated with a zero. We now primarily represent OTU tables in a sparse format: if an OTU is not observed in a sample, there is no count for that OTU. The two ways of representing this data are exemplified here.</p>
<p>A dense representation of an OTU table:</p>
<div class="highlight-python"><pre>OTU ID PC.354  PC.355  PC.356
OTU0   0   0   4
OTU1   6   0   0
OTU2   1   0   7
OTU3   0   0   3</pre>
</div>
<p>A sparse representation of an OTU table:</p>
<div class="highlight-python"><pre>PC.354 OTU1 6
PC.354 OTU2 1
PC.356 OTU0 4
PC.356 OTU2 7
PC.356 OTU3 3</pre>
</div>
<p>OTU table data tends to be sparse (e.g., greater than 90% of counts are zero, and frequently as many as 99% of counts are zero) in which case the latter format is more convenient to work with as it has a smaller memory footprint. Both of these representations are supported in the biom-format project via dense and sparse Table types. Generally if less than 85% of your counts are zero, a dense representation will be more efficient.</p>
</div>
<div class="section" id="encapsulation-of-core-study-data-otu-table-data-and-sample-otu-metadata-in-a-single-file">
<h3>Encapsulation of core study data (OTU table data and sample/OTU metadata) in a single file<a class="headerlink" href="#encapsulation-of-core-study-data-otu-table-data-and-sample-otu-metadata-in-a-single-file" title="Permalink to this headline">¶</a></h3>
<p>The JSON-format OTU table allow for storage of arbitrary amounts of sample and OTU metadata in a single file. Sample metadata corresponds to what is generally found in QIIME mapping files. At this stage inclusion of this information in the OTU table file is optional, but it may be useful for sharing these files with other QIIME users and for publishing or archiving results of analyses. OTU metadata (generally a taxonomic assignment for an OTU) is also optional. In contrast to the previous OTU table format, you can now store more than one OTU metadata value in this field, so for example you can score taxonomic assignments based on two different taxonomic assignment approaches.</p>
</div>
<div class="section" id="facilitating-the-use-of-tables-between-tools-that-support-this-format">
<h3>Facilitating the use of tables between tools that support this format<a class="headerlink" href="#facilitating-the-use-of-tables-between-tools-that-support-this-format" title="Permalink to this headline">¶</a></h3>
<p>Different tools, such as <a class="reference external" href="http://www.qiime.org">QIIME</a>, <a class="reference external" href="http://metagenomics.anl.gov">MG-RAST</a>, and <a class="reference external" href="http://vamps.mbl.edu/">VAMPS</a> work with similar data structures that represent different types of data. An example of this is a <cite>metagenome</cite> table that could be generated by MG-RAST (where for example, columns are metagenomes and rows are functional categories). Exporting this data from MG-RAST in a suitable format will allow for the application of many of the QIIME tools to this data (such as generation of alpha rarefaction plots or beta diversity ordination plots). This new format is far more general than previous formats, so will support adoption by groups working with different data types and is already being integrated to support transfer of data between <a class="reference external" href="http://www.qiime.org">QIIME</a>, <a class="reference external" href="http://metagenomics.anl.gov">MG-RAST</a>, and <a class="reference external" href="http://vamps.mbl.edu/">VAMPS</a>.</p>
</div>
</div>
<div class="section" id="file-extension">
<h2>File extension<a class="headerlink" href="#file-extension" title="Permalink to this headline">¶</a></h2>
<p>We recommend that BIOM files use the <tt class="docutils literal"><span class="pre">.biom</span></tt> extension.</p>
</div>
</div>


      </div>
      <div class="bottomnav">
      
        <p>
        «&#160;&#160;<a href="index.html">BIOM Documentation</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="../index.html">Contents</a>
        &#160;&#160;::&#160;&#160;
        <a href="table_objects.html">biom-format <tt class="docutils literal docutils literal"><span class="pre">Table</span></tt> objects</a>&#160;&#160;»
        </p>

      </div>


    <div class="footer">
        &copy; Copyright 2012, The BIOM Project.
      Last updated on May 01, 2012.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.0.7.
    </div>
<script type="text/javascript">

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-6636235-6']);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();

</script>

  </body>
</html>